### PR TITLE
add meson build system

### PR DESCRIPTION
**Pull request by kusma**
Meson is quite a bit faster than Autotools, which seems like a good
reason to try this out:

Full rebuild, autotools:
> real 300.18
> user 235.32
> sys 32.37

Full rebuild, meson:
> real 46.92
> user 70.71
> sys 11.24

The time above is the time './configure && make', and
'meson . build && ninja -C build' took on my Toshiba AC100, measured
using '/usr/bin/time -p'.

So, less than a sixth of the time for a full rebuild, which is very
neat. However, it's on partial rebuilds that it really shines:

Partial rebuild, autotools:
> real 30.60
> user 21.07
> sys 4.27

Partial rebuild, meson:
> real 2.27
> user 1.92
> sys 0.27

In this case, I touched src/libhost1x/host1x-gr3d.c, and rebuilt
with 'make' and 'ninja' respectively. And the result is pretty
awesome, dropping to below a tenth of the build-time.

### CODE REVIEW

**Comment by kusma** [comment]
The only things I know of that this is lacking compared to the autotools-system is:
- Generating pkgonfig-files. This is [not very hard to do](http://mesonbuild.com/Pkg-config-files.html), but I'm not very convinced it's really nessecary. @digetx, you added the pkgconfig-file generation initially... what do you use it for?
- The ability to opt-out of building GLES-tests. Since meson is so much faster, I didn't feel like this was nessecary to support. It also wouldn't be the hardest thing ever to add if needed.

I'm also not (yet?) proposing we remove the automake-files. I'd rather we test this out a bit in-tree before we do something drastic ;)

**Comment by digetx** [comment]
I was experimenting with using libgrate for opentegra and haven't used pkgconfig since then, yeah it is not really needed.

Ninja is indeed much faster than automake, but it's not a problem for me since I'm crosscompiling.  I'll probably continue to use autotools, so have no objections ;) I made a test build with meson, works fine 👍

File: tools/meson.build [hunk 1]
```diff
@@ -0,0 +1,23 @@
+tests = [
```
Comment (digetx) [inline tools/meson.build]: s/tests/tools/ ?

Comment (kusma) [reply tools/meson.build]: Whops, yes! Good catch!

**Comment by kusma** [comment]
@digetx: Yeah, after trying and failing a bit to try to get a workable cross-compiling setup, I decided to just try to make my existing workflow better instead ;)

**Comment by kusma** [comment]
@digetx: updated with some cleanups:
- fixed the typo you pointed out (tests -> tools etc)
- switched to implicit rules for parsers/lexers

To do the latter, I had to rename the sources. It's a bit boring, but matching artifact-names and source names seems reasonable to me, so whatever ;)

**Review by digetx — APPROVED** [review]
Good work! 👍

**Pull request merged**